filter.js

```diff
@@ -0,0 +1,168 @@
+  const online = document.querySelector("#online");
+  const onsite = document.querySelector("#onsite");
+  
+  // Functions to handle type click
+  online.addEventListener("click", function () {
+    if (filter.online == true) {
+      filter.online = false;
+    } else {
+      filter.online = true;
+    }
+  });
+  
+  onsite.addEventListener("click", function () {
+    if (filter.onsite == true) {
+      filter.onsite = false;
+    } else {
+      filter.onsite = true;
+    }
+  });
+  
+  /*------------------------BY TYPE------------------------*/
+  
+  /*------------------------STAR RATING------------------------*/
+  
+  // Get all the stars
+  const fromStars = [
+    document.querySelector("#fromOne"),
+    document.querySelector("#fromTwo"),
+    document.querySelector("#fromThree"),
+    document.querySelector("#fromFour"),
+    document.querySelector("#fromFive"),
+  ];
+  
+  const toStars = [
+    document.querySelector("#toOne"),
+    document.querySelector("#toTwo"),
+    document.querySelector("#toThree"),
+    document.querySelector("#toFour"),
+    document.querySelector("#toFive"),
+  ];
+  
+  // Add event listeners to all stars
+  fromStars.forEach((star, index) => {
+    star.addEventListener("click", () => handleStarClick(fromStars, index, true));
+  });
+  
+  toStars.forEach((star, index) => {
+    star.addEventListener("click", () => handleStarClick(toStars, index, false));
+  });
+  
+  // Function to handle star click
+  function handleStarClick(stars, index, isFromStars) {
+    // Remove 'checked' class from all stars
+    stars.forEach((star) => star.classList.remove("checked"));
+  
+    // Add 'checked' class to clicked star and all previous stars
+    for (let i = 0; i <= index; i++) {
+      stars[i].classList.add("checked");
+    }
+  
+    // Update the rating
+    const rating = index + 1;
+    if (isFromStars) {
+      // Only update minRating if it's less than or equal to maxRating
+      if (rating <= filter.maxRating) {
+        filter.minRating = rating;
+      } else {
+        // If minRating would be greater than maxRating, remove 'checked' class from the stars that should not be selected
+        for (let i = filter.maxRating; i <= index; i++) {
+          stars[i].classList.remove("checked");
+        }
+      }
+    } else {
+      filter.maxRating = rating;
+    }
+  }
+  
+  /*------------------------STAR RATING------------------------*/
+  
+  /*------------------------TAGS------------------------*/
+  
+  // Get all the tags
+  
+  const tags = [
+    document.querySelector("#web"),
+    document.querySelector("#linux"),
+    document.querySelector("#cryptography"),
+    document.querySelector("#coding"),
+    document.querySelector("#ssh"),
+    document.querySelector("#ctf"),
+    document.querySelector("#hacking"),
+    document.querySelector("#bash"),
+    document.querySelector("#javascript"),
+    document.querySelector("#electronics"),
+    document.querySelector("#phreaking"),
```

Comment:
Hard coding the labels in the HTML and retrieving them like this means that it's impossible to add new labels without changing HTML code. Getting them from the API would have been better.